packages/shared/src/components/modals/post/CommentModal.tsx

```diff
@@ -0,0 +1,262 @@
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const item of data?.postComments?.edges) {
+    if (item.node.id === commentId) {
+      return item.node;
+    }
+    if (item.node.children) {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const child of item.node.children.edges) {
+        if (child.node.id === commentId) {
+          return child.node;
+        }
+      }
+    }
+  }
+
+  return undefined;
+};
+
+// statically defined heights of various containers that
+// affect final height of the input container
+// when styles change, these should be updated accordingly
+const headerSize = 57;
+const replySize = 40;
+const footerSize = 48;
+
+export interface CommentModalProps
+  extends LazyModalCommonProps,
+    CommentMarkdownInputProps {
+  replyToCommentId?: string;
+}
+
+export default function CommentModal({
+  isOpen,
+  onRequestClose,
+  onAfterClose,
+  onCommented,
+  parentCommentId,
+  replyToCommentId,
+  editCommentId,
+  post,
+}: CommentModalProps): ReactElement {
+  const [value, setValue] = useState('');
+  const [styleHeight, setStyleHeight] = useState<number | 'auto'>('auto');
+  const { user } = useAuthContext();
+  const client = useQueryClient();
+  const [modalNode, setModalNode] = useState<HTMLElement>(null);
+
+  const isEdit = !!editCommentId;
+  const isReply = !isEdit && !!replyToCommentId;
+
+  const refCallback = useCallback(
+    (node) => {
+      if (node) {
+        setModalNode(node);
+      }
+    },
+    [setModalNode],
+  );
+
+  const { shouldShowCta, isEnabled, onToggle, onSubmitted } =
+    useNotificationToggle({
+      source: NotificationPromptSource.NewComment,
+    });
+
+  const onSuccess: typeof onCommented = (comment, isNew, parentCommentID) => {
+    if (onCommented) {
+      onCommented(comment, isNew, parentCommentID);
+    }
+    onRequestClose();
+  };
+
+  const comment = useMemo(
+    () =>
+      getCommentFromCache({
+        client,
+        postId: post?.id,
+        commentId: isEdit ? editCommentId : replyToCommentId ?? parentCommentId,
+      }),
+    [
+      client,
+      post?.id,
+      isEdit,
+      editCommentId,
+      parentCommentId,
+      replyToCommentId,
+    ],
+  );
+
+  const { mutateComment, isLoading } = useMutateComment({
+    post,
+    editCommentId: isEdit && editCommentId,
+    parentCommentId: isEdit ? comment?.parent?.id : parentCommentId,
+    onCommented: onSuccess,
+  });
+
+  useLayoutEffect(() => {
+    // scroll to bottom of modal
+    modalNode?.scrollTo({ behavior: 'auto', top: 10000 });
+  }, [modalNode]);
+
+  const { height } = useVisualViewport();
+  const replyHeight = height > 0 ? replySize : 0;
+  const footerHeight = shouldShowCta ? footerSize : 0;
+  const totalHeight = height - headerSize - replyHeight - footerHeight;
+  const inputHeight = totalHeight > 0 ? Math.max(totalHeight, 300) : 'auto';
+
+  useEffect(() => {
+    setStyleHeight(inputHeight);
+  }, [inputHeight, setStyleHeight]);
```

Comment:
Don't think you need this actually?
Can just be outside the effect no?

Reply:
Refactored, the new implementation is setting the style prop directly on the current ref instead.